### ISSUE TITLE
confmap Resolver URI Parsing doesn't work for valid yaml URIs

### ISSUE THREAD

**Issue by cpheps** [open-telemetry/opentelemetry-collector#6306]
**Describe the bug**
With the URI  parsing changes from #6271 in the `confgmap.Resolver` valid yaml URIs no longer work. 

**Steps to reproduce**
Take a valid yaml config string and use similar logic to below will result in an invalid URI error:

```go
uriLocation := fmt.Sprintf("yaml:%s", yamlBytes)
provider := yamlprovider.New()
converter := expandconverter.New()
settings := service.ConfigProviderSettings{
	ResolverSettings: confmap.ResolverSettings{
		URIs:       []string{uriLocation},
		Providers:  map[string]confmap.Provider{provider.Scheme(): provider},
		Converters: []confmap.Converter{converter},
	},
}

service.NewConfigProvider(settings)
```

**What did you expect to see?**
No URI error

**What did you see instead?**
Error for invalid URI

**What version did you use?**
Version: v0.62.0

**What config did you use?**
any valid config

**Environment**
OS: macOS 12
Compiler(if manually compiled): go 1.19

**Additional context**
This seems to be an issue with the `locationRegexp`.  I took a valid yaml URI from the yaml provider test [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/provider/yamlprovider/provider_test.go#L97) and plugged it into the regex and it fails to parse. I have this up on the go playground [here](https://go.dev/play/p/YFU8hdwMyM4).

Looks like it is due to the newlines in the yaml. If you remove the newlines the URI works or if you remove the `$` from the `locationRegexp` it also works.

**Comment by cpheps** [comment]
Investigating further it seems this regex will work for yaml URIs and other kinds.

```
^(?P<Scheme>[A-Za-z][A-Za-z0-9+.-]+):(?P<OpaqueValue>[\s\S]*)$
```

**Comment by bogdandrutu** [comment]
found the issue, give me 2 mins

**Comment by cpheps** [comment]
Fixed by #6309
